### PR TITLE
Feat 2105 combo airdrop

### LINKED ISSUE #2022: Add lido to the list of detected airdrops
## Abstract

DO of lido.fi is an airdropped token 

@bantg sent me a list here: https://github.com/vshvsh/ldo-airdrop/blob/master/snapshot/lido-07-balances.json

It's not on coingecko yet since it's not tradeable but perhaps we will wait until it is to be able to implement it.

## Task

Do it

### LINKED ISSUE #2105: Add the COMBO token to the airdrops
## Abstract

Furucombo has launched a token and airdropped to their users and donors on January 15th. 
These tokens can be claimed periodically over a vesting period of 8 weeks.

## Motivation

The airdrop ranges with either 350 or 900 COMBO which at the time of writing it worth 560 and 1450 usd. It would be nice have a easy way to track these tokes so the user can claim them.

## Specification
Add the COMBO token to the airdrop section and update the balance of the COMBO tokens in each vesting cycle.
Each vesting cycle is at every friday for 8 weeks afaik.   
Idk how much will be distributed each cycle but those who had 350 where able to unlock 87.5 today.   

A list of eligible addresses can be found here:
https://docs.google.com/spreadsheets/d/113AiPrGJ-yp7g-Kdo_IofMWiftc7msEoBMmKNKxJ4fo/edit#gid=0
The announcement of the token airdrop can be found here: 
https://medium.com/furucombo/first-furucombo-grant-7b1e48175c99

### PR DESCRIPTION
**Pull request by vnavascues**
Closes #2022 
Closes #2105 

[Airdrop original document](https://docs.google.com/spreadsheets/d/113AiPrGJ-yp7g-Kdo_IofMWiftc7msEoBMmKNKxJ4fo/edit#gid=0)
[Airdrop processed document](https://gist.githubusercontent.com/vnavascues/5f32d53a7ebd1ea844f10cb48e7b7002/raw/fa0c947c6dd1ee2a7b16674a86f0e50cb7bbe631/furucombo_airdrop.csv) (we may want to store it as a Lefteris gist)

Rationale:
- There are 3 types of data under "Full List": user names (Combo Master type), L1 ethereum addresses and L2 ethereum addresses. [Furucombo medium post says](https://medium.com/furucombo/first-furucombo-grant-7b1e48175c99). Both usernames and L2 addresses have been discarded from the processed gist. In fact L2 have been advised to:

> If you donated to Furucombo with ZKSync and find yourself on the list with a L2 tag, please contact us on discord to verify your L1 address.

- COMBO airdrop starts on the 2021-01-15 at 03:00:00 UTC with an amount that comes from the `Week 0` header. Every Friday at 3:00 (+7 natural days) a new amount is vested (from `Week 1`, `Week 2`... headers). We return this amount during the whole week until next vesting point. After 9 natural weeks (a week past after the last vesting point, from `Week 8` header) we do not return anymore COMBO amounts.

### CODE REVIEW

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 1]
```diff
@@ -74,17 +100,34 @@ def check_airdrops(
...
+        if protocol_name == 'furucombo':
+            if timestamp_now < FURUCOMBO_AIRDROP_ENDS_AT:  # do not display after week 9 finishes
+                for row in data:
+                    addr = row[0]
+                    if addr in addresses:
+                        amount = get_furucombo_claimable_combo(
+                            row=row,
+                            timestamp_now=timestamp_now,
+                        )
+                        found_data[addr][protocol_name] = {
```
Comment (vnavascues) [inline rotkehlchen/chain/ethereum/airdrops.py:113]: Vesting schemas are tricky and definitely we don't want to be very specific (front-end work). But we could put in "amount" the "total amount" (e.g. 350, 900 COMBO, we have it and generally all airdrops include it) and then add an extra "info" (or "details") text attribute where we can put the amount in context and advise the user. For instance: "This address is eligible to claim 350 COMBO that will be vested over eight weeks. Please, check the Furucombo airdrop details.".

Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/airdrops.py:113]: we will just put total for now as we do for all. This is not a balance return, but how much they can get

Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/airdrops.py:113]: > COMBO airdrop starts on the 2021-01-15 at 03:00:00 UTC with an amount that comes from the Week 0 header. Every Friday at 3:00 (+7 natural days) a new amount is vested (from Week 1, Week 2... headers). We return this amount during the whole week until next vesting point. After 9 natural weeks (a week past after the last vesting point, from Week 8 header) we do not return anymore COMBO amounts.

Why nothing after week 9? Does it stop being claimable? I don't see anything like that in the medium post.

File: rotkehlchen/data/all_assets.meta [hunk 2]
```diff
@@ -1 +1 @@
-{"md5": "7969237a9e90ff27d35c29e2ff2fe895", "version": 47}
```
Comment (LefterisJP) [inline rotkehlchen/data/all_assets.meta:1]: you are going to add a token upgrade anyway for other assets later. So don't bump the version yet. Do it after all token changes are done.

File: rotkehlchen/data/all_assets.json [hunk 3]
```diff
@@ -14999,5 +14999,14 @@
         "started": 1604214421,
         "symbol": "ROOK",
         "type": "ethereum token"
+    },
+    "LDO": {
+        "coingecko": "lido-dao",
+        "ethereum_address": "0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32",
+        "ethereum_token_decimals": 18,
+        "name": "Lido DAO",
```
Comment (LefterisJP) [inline rotkehlchen/data/all_assets.json:15007]: Name from the contract is: `Lido DAO Token`

**Comment by LefterisJP** [comment]
Oh also this would need rebase on develop and changelog additions for both the Lido token and the airdrops support.

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 4]
```diff
@@ -39,7 +42,21 @@
         A_GRAIN,
         'https://claim.harvest.finance/',
     ),
+    'furucombo': (
+        # is checksummed
+        'https://gist.githubusercontent.com/vnavascues/5f32d53a7ebd1ea844f10cb48e7b7002/raw/fa0c947c6dd1ee2a7b16674a86f0e50cb7bbe631/furucombo_airdrop.csv',  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/airdrops.py:46]: Same CSV just hosted by my user: https://gist.githubusercontent.com/LefterisJP/69612e155e8063fd6b3422d4efbf22a3/raw/b9023960ab1c478ee2620c456e208e5124115c19/furucombo_airdrop.csv

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 5]
```diff
@@ -39,7 +42,21 @@
...
+        A_COMBO,
+        'https://furucombo.app/',
+    ),
+    'lido': (
+        # is checksummed
+        'https://gist.githubusercontent.com/vnavascues/6fea5eb631af7e0b8a910e8c62cd471e/raw/fa529e9cc5a90e0970f59f69c649e45b2008bbf9/lido_airdrop.csv',  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/airdrops.py:52]: Same CSV just hosted by my user: https://gist.githubusercontent.com/LefterisJP/57a8d65280a482fed6f3e2cc00c0e540/raw/e6ebac56c438cc8a882585c5f5bfba64eb57c424/lido_airdrop.csv

**Review by LefterisJP — COMMENTED** [review]
So you removed the furucombo logic? Also there is no changelog entry neither for the 2 airdrops or the lido dao token addition.

**Comment by LefterisJP** [comment]
Hey @kelsos here e2e failed but they passed on my PR which is based on this. So it's probably something one off?

**Review by LefterisJP — APPROVED** [review]
LGTM. I went through all test failures. All related to kraken being down.

**Pull request merged**